test/integration/camera_sensor.cc

```diff
@@ -2268,3 +2268,151 @@ TEST_F(CameraSensor, CheckNewAndLegacyDistortionModes)
+  unsigned int defaultCamWidth = defaultIntrinsicsCam->ImageWidth();
+  unsigned int defaultCamHeight = defaultIntrinsicsCam->ImageHeight();
+  double defaultCamFx = defaultIntrinsicsCam->ImageFocalLengthX();
+  double defaultCamFy = defaultIntrinsicsCam->ImageFocalLengthY();
+  double defaultCamCx = defaultIntrinsicsCam->ImageOpticalCenterX();
+  double defaultCamCy = defaultIntrinsicsCam->ImageOpticalCenterY();
+
+  EXPECT_EQ(defaultCamWidth, 320u);
+  EXPECT_EQ(defaultCamHeight, 240u);
+  double error = 0.0005;
+  EXPECT_NEAR(defaultCamFx, 277.127, error);
+  EXPECT_NEAR(defaultCamFy, 277.127, error);
+  EXPECT_DOUBLE_EQ(defaultCamCx, 160);
+  EXPECT_DOUBLE_EQ(defaultCamCy, 120);
+
+  // get the 'intrinsics_camera' sensor, <intrinsics> tag is used explicitly for this sensor
+  // where the intrinsics provided are same as gazebo default.
+  sensors::SensorPtr intrinsicsSensor = sensors::get_sensor("intrinsics_camera_sensor");
+  sensors::CameraSensorPtr intrinsicsCamSensor =
+      std::dynamic_pointer_cast<sensors::CameraSensor>(intrinsicsSensor);
+  EXPECT_TRUE(intrinsicsCamSensor != nullptr);
+  rendering::CameraPtr intrinsicsCam = intrinsicsCamSensor->Camera();
+  EXPECT_TRUE(intrinsicsCam != nullptr);
+
+  // image size, focal length and optical centre for 'intrinsics_camera' sensor
+  unsigned int  intrinsicsCamWidth = intrinsicsCam->ImageWidth();
+  unsigned int  intrinsicsCamHeight = intrinsicsCam->ImageHeight();
+  double intrinsicsCamFx = intrinsicsCam->ImageFocalLengthX();
+  double intrinsicsCamFy = intrinsicsCam->ImageFocalLengthY();
+  double intrinsicsCamCx = intrinsicsCam->ImageOpticalCenterX();
+  double intrinsicsCamCy = intrinsicsCam->ImageOpticalCenterY();
+  EXPECT_EQ(intrinsicsCamWidth, 320u);
+  EXPECT_EQ(intrinsicsCamHeight, 240u);
+  EXPECT_DOUBLE_EQ(intrinsicsCamFx, 277);
+  EXPECT_DOUBLE_EQ(intrinsicsCamFy, 277);
+  EXPECT_DOUBLE_EQ(intrinsicsCamCx, 160);
+  EXPECT_DOUBLE_EQ(intrinsicsCamCy, 120);
+
+  // get the 'camera' sensor, <intrinsics> tag is used explicitly for this sensor
+  // where the intrinsics provided are different from the gazebo default.
+  sensors::SensorPtr sensor = sensors::get_sensor("camera_sensor");
+  sensors::CameraSensorPtr camSensor =
+      std::dynamic_pointer_cast<sensors::CameraSensor>(sensor);
+  EXPECT_TRUE(camSensor != nullptr);
+  rendering::CameraPtr cam = camSensor->Camera();
+  EXPECT_TRUE(cam != nullptr);
+
+  // image size, focal length and optical centre for 'camera' sensor
+  unsigned int camWidth = cam->ImageWidth();
+  unsigned int camHeight = cam->ImageHeight();
+  double camFx = cam->ImageFocalLengthX();
+  double camFy = cam->ImageFocalLengthY();
+  double camCx = cam->ImageOpticalCenterX();
+  double camCy = cam->ImageOpticalCenterY();
+  EXPECT_EQ(camWidth, 320u);
+  EXPECT_EQ(camHeight, 240u);
+  EXPECT_DOUBLE_EQ(camFx, 320);
+  EXPECT_DOUBLE_EQ(camFy, 320);
+  EXPECT_DOUBLE_EQ(camCx, 160);
+  EXPECT_DOUBLE_EQ(camCy, 120);
+
+  // connect to new frame event
```

Comment:
Could you capitalize the fist letters ? `` // Connect to new frame event.``

Reply:
👍